internal/command/jsonformat/change/map.go

```diff
@@ -0,0 +1,83 @@
+package change
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+
+	"github.com/hashicorp/terraform/internal/command/format"
+	"github.com/hashicorp/terraform/internal/plans"
+)
+
+func Map(elements map[string]Change) Renderer {
+	maximumKeyLen := 0
+	for key := range elements {
+		if maximumKeyLen < len(key) {
+			maximumKeyLen = len(key)
```

Comment:
Comparing this to the equivalent for objects, I'm curious about (a) why we don't take into account string escaping here and (b) why we calculate the maximum length in this constructor instead of the `Render` method.

Reply:
One difference between objects and maps is that maps always render the key surrounded by quotation marks. So for maps the key escaping doesn't really matter, it's always escaped and that means the relative length is always `+2` so calculating the length with or without the quotation marks doesn't really matter as the math still works when adding in the additional padding.

I did this in the constructor on the theory that the keys never change, therefore the maximum length never changes, therefore lets calculate the maximum length once at the beginning.

For objects, I actually missed that functionality around sometimes they need to be escaped, so the object calculations change in a later PR (one not sent for review yet), that do move this out of the constructor for objects at the very least.